src/core/scopes_configs.cc

```diff
@@ -623,6 +623,23 @@ int32_t scopes_configs::generate_alert(waflz_pb::alert** ao_alert,
         // -------------------------------------------------
         l_at->mutable_req_info()->set_customer_id(a_cust_id);
         // -------------------------------------------------
+        // set account_type
+        // -------------------------------------------------
+        cust_id_scopes_map_t::iterator i_scopes;
+        if (i_scopes != m_cust_id_scopes_map.end())
+        {
+                i_scopes = m_cust_id_scopes_map.find(a_cust_id);
+                waflz_pb::limit *l_limit = l_at->mutable_limit();
+                if(i_scopes->second->get_pb()->has_account_type()){
+                        std::string l_account_type = i_scopes->second->get_account_type();
+                        if (l_account_type.empty()){
+                        }else{
+                                NDBG_PRINT("account_type not found\n");
+                                l_limit->set_account_type("__na__");
```

Comment:
you can do this in one liner `l_limit->set_account_type(i_scopes->second->get_account_type()`. get_account_type will return `m_account_type`. Initialize that to `__na__`. This will save you extra string copy and you need if, else